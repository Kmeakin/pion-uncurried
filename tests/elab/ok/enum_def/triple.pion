enum Triple(A: Type, B: Type, C: Type) {
    MkPair(x: A, y: B),
}

let mkpair = MkPair;

let fst 
= fn(A: Type, B: Type, C: Type, p: Triple(A, B, C)) => match p {
    MkPair(z, foo) => z,
};


let fst_ann 
: fn (X: Type, Y: Type, Z: Type, q: Triple(X, Y, Z)) -> X
= fn(A: Type, B: Type, C: Type, p: Triple(A, B, C)) => match p {
    MkPair(z, foo) => z,
};

