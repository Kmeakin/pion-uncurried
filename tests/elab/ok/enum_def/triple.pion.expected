enum Triple(A: Type, B: Type, C: Type): Type {
    MkPair(x: A, y: B): Triple(A, B, C),
}

let mkpair: fn(A: Type, B: Type, C: Type, x: A, y: B) -> Triple(A, B, C) = MkPair;

let fst: fn(A: Type, B: Type, C: Type, p: Triple(A, B, C)) -> A = fn(A: Type, B: Type, C: Type, p: Triple(A, B, C)) => match p {
    MkPair(z, foo) => z,
};

let fst_ann: fn(X: Type, Y: Type, Z: Type, q: Triple(X, Y, Z)) -> X = fn(A: Type, B: Type, C: Type, p: Triple(A, B, C)) => match p {
    MkPair(z, foo) => z,
};
